src/main/java/net/snowflake/client/core/QueryContextElement.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.client.core;
+
+import java.util.Arrays;
+
+/** Query context information. */
+class QueryContextElement implements Comparable<QueryContextElement> {
```

Comment:
I would make this class static private of the QueryContextCache because the cache manipulates it directly. 
Nobody should / would use those elements outside of the cache

Reply:
Done. thanks for your comments.